### ISSUE TITLE
Stop raising FormDataRoutingRedirect

### ISSUE THREAD

**Issue by davidism** [pallets/flask#3003]
https://github.com/pallets/werkzeug/pull/1342 sets the status code for `RequestRedirect` to 308, so the request body (form data) will be preserved. We no longer need to fail with `FormDataRoutingRedirect`. @pgjones

**Comment by theodesp** [comment]
Hey can I work on this? As I can see here:
https://github.com/pallets/flask/blob/20f6040ed7680c86543f9cd4f01ca6dcc6eaf27e/flask/app.py#L1775-L1776

We only dispatch this on debug mode so we can safely remove. Is that OK?

**Comment by davidism** [comment]
Yeah, go for it! Can't completely remove it because we'd have to bump the minimum Werkzeug version to 0.15, which hasn't been released yet. For now it should check if the status code is 307 or 308 and not raise in that case.

**Comment by ThiefMaster** [comment]
I think in debug mode we should still fail. Triggering a slashes-related redirect on form submission likely indicates a bug or bad code (e.g. not using `url_for`)

**Comment by davidism** [comment]
Hmm, yeah. Probably best to leave it as people should really fix the URL. Although what about `alias` rules?
